firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -146,14 +146,22 @@ private void preConditionChecks() {
     Preconditions.checkNotEmpty(getApplicationId());
     Preconditions.checkNotEmpty(getProjectIdentifier());
     Preconditions.checkNotEmpty(getApiKey());
+    Preconditions.checkArgument(
+            Utils.isValidAppIdFormat(getApplicationId()),
+            "Please set your Application ID. A valid Firebase App ID is required to communicate "
+                    + "with Firebase server APIs: It identifies your application with Firebase."
+                    + "Please refer to https://firebase.google.com/support/privacy/init-options.");
+    Preconditions.checkArgument(
+            Utils.isValidApiKeyFormat(getApiKey()),
+            "Please set a valid API key. A Firebase API key is required to communicate with "
+                    + "Firebase server APIs: It authenticates your project with Google."
+                    + "Please refer to https://firebase.google.com/support/privacy/init-options.");
   }
 
   /** Returns the Project Id or Project Number for the Firebase Project. */
   @Nullable
   String getProjectIdentifier() {
-    return TextUtils.isEmpty(firebaseApp.getOptions().getProjectId())
-        ? firebaseApp.getOptions().getGcmSenderId()
-        : firebaseApp.getOptions().getProjectId();
+    return firebaseApp.getOptions().getProjectId();
```

Comment:
Is there no test that you have to modify?
(I mean a test that created FirebaseApp without projectId but with gcmSenderId.)

Reply:
No, we don't have